src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/aggregator/CategoricalProportionAggregator.java

```diff
@@ -42,9 +53,16 @@ public boolean appliesTo(String[] rec) {
 
   @Override
   public Double finish() {
-    if (numDenominatorMatches == 0) {
+    int totalNumeratorMatches;
+    if (negationMode) {
+      totalNumeratorMatches = numNonNullMatches - numNumeratorMatches;
+    } else {
+      totalNumeratorMatches = numNumeratorMatches;
+    }
+    final int totalDenominatorMatches = totalNumeratorMatches + distinctDenominatorMatchCount;
```

Comment:
This seems weird to me, but don't really understand what you're going for.  We could easily be double counting here.  If distinctDenominatorMatchCount is "in denominator set but not in numerator set" (i.e. distinct from numerator matches, then shouldn't you have logic above that does an "else if" after the numeratorMatchSet check?